### ISSUE TITLE
Cicero outputting {{/optional}} in contract output

### ISSUE THREAD

**Issue by niallroche** [accordproject/cicero#625]
# Bug Report 🐛
where using an {{#optional block with an else in either the true or false case of the dependent variable the trailing close {{/optional}} is being output into the contract text

## Expected Behavior
the contents of the optional or else blocks only should be output

## Current Behavior
{{/optional}} is being appended

## Steps to Reproduce

[sampledata_sections.json.txt](https://github.com/accordproject/cicero/files/6190463/sampledata_sections.json.txt)
[sampledata_works.json.txt](https://github.com/accordproject/cicero/files/6190464/sampledata_works.json.txt)
[grammar.tem.md.txt](https://github.com/accordproject/cicero/files/6190466/grammar.tem.md.txt)
[outputfrom_sampledata_sections2.md.txt](https://github.com/accordproject/cicero/files/6190473/outputfrom_sampledata_sections2.md.txt)
[model.cto.txt](https://github.com/accordproject/cicero/files/6190474/model.cto.txt)

### Desktop
 - OS: macOS

**Comment by jeromesimeon** [comment]
This issue is more or less a duplicate of https://github.com/accordproject/markdown-transform/issues/316

The example is not currently supported (you cannot have an `{{optional}}` that spans multiple commonmark [blocks](https://spec.commonmark.org/0.29/#blocks).

You can see why there is an `{{/optional}}` left in the templatemark dingus here: https://spec.commonmark.org/0.29/#blocks

If you inspect the AST there, you will see that the optional is (automatically) closed at the end of the first paragraph.

**Comment by jeromesimeon** [comment]
This is true of all "inline blocks" in the documentation: https://docs.accordproject.org/docs/markup-templatemark.html#inline-blocks
